pkg/services/updatechecker/instrumentation/httpclient.go

```diff
@@ -0,0 +1,128 @@
+package instrumentation
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/codes"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/tracing"
+)
+
+// InstrumentedHTTPClient is an HTTP client that wraps every request with a span, tracking the method, url,
+// and response code as span attributes.
+type InstrumentedHTTPClient struct {
```

Comment:
This definitely works. But would consider reusing the SDK httpclient library we already have that supports middlewares. Using [httpclient.New](https://github.com/grafana/grafana-plugin-sdk-go/blob/4344c93ea2747c85a2d3d18635be0766e56af4be/backend/httpclient/http_client.go#L18). Then maybe this [tracing middleware](https://github.com/grafana/grafana/blob/f0333ac41f0d2b0c0107f5794b7d5c7ca8aa0401/pkg/infra/httpclient/httpclientprovider/tracing_middleware.go#L21) can be resued?

WDYT?

Reply:
Switched to httpclient tracing middleware and a new prometheus metrics middleware as suggested :)